### PR TITLE
{Storage} `az storage file delete/resize/url/generate-sas/show/update/exists`: Track2 SDK migration

### PR DESCRIPTION
**Pull request by calvinhzy**
**Related command**

**Description**

Migrate to track2 sdk 
`az storage file delete/resize/url/generate-sas/show/update/exists`

**Testing Guide**

`az storage file delete -p {filepath} -s {sharename}`
`az storage file resize -p {filepath} -s {sharename} --size 128`
`az storage file url -p {filepath} -s {sharename}`
`az storage file generate-sas -p {filepath} -s {sharename} --start 2022-01-01 --expiry 2022-12-31 --permission rw`
`az storage file show -p {filepath} -s {sharename}`
`az storage file exists -p {filepath} -s {sharename}`

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Storage

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 1]
```diff
@@ -1935,6 +1935,7 @@
 
 helps['storage file generate-sas'] = """
 type: command
+short-summary: Generates a shared access signature for the file.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:1938]:
```suggestion
short-summary: Generate a shared access signature for the file.
```

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 2]
```diff
@@ -1950,6 +1951,15 @@
     crafted: true
 """
 
+helps['storage file show'] = """
+type: command
+short-summary: Returns all user-defined metadata, standard HTTP properties, and system properties for the file.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:1956]:
```suggestion
short-summary: Return all user-defined metadata, standard HTTP properties, and system properties for the file.
```

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 3]
```diff
@@ -1964,11 +1974,37 @@
     crafted: true
 """
 
+helps['storage file delete'] = """
+type: command
+short-summary: Marks the specified file for deletion.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:1979]:
```suggestion
short-summary: Mark the specified file for deletion.
```

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 4]
```diff
@@ -1964,11 +1974,37 @@
...
+long-summary: The file is later deleted during garbage collection.
+"""
+
+helps['storage file resize'] = """
+type: command
+short-summary: Resizes a file to the specified size.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:1985]:
```suggestion
short-summary: Resize a file to the specified size.
```

File: src/azure-cli/azure/cli/command_modules/storage/_help.py [hunk 5]
```diff
@@ -1964,11 +1974,37 @@
...
+"""
+
 helps['storage file metadata'] = """
 type: group
 short-summary: Manage file metadata.
 """
 
+helps['storage file update'] = """
+type: command
+short-summary: Sets system properties on the file.
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/_help.py:2000]:
```suggestion
short-summary: Set system properties on the file.
```

File: src/azure-cli/azure/cli/command_modules/storage/operations/file.py [hunk 6]
```diff
@@ -407,9 +414,34 @@ def _make_directory_in_files_share(file_service, file_share, directory_path, exi
...
+    return t_generate_file_sas(account_name=client.account_name, share_name=client.share_name, file_path=file_path,
+                               account_key=client.credential.account_key, permission=permission, expiry=expiry,
+                               start=start, policy_id=id, ip=ip, protocol=protocol, **kwargs)
+
+
+def file_exists(client, **kwargs):
+    try:
+        res = client.get_file_properties(**kwargs)
+        return bool(res)
+    except HttpResponseError:
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/operations/file.py:442]: Better to return False for `ResourceNotFoundError` and raise if it's other `HttpResponseError`

File: src/azure-cli/azure/cli/command_modules/storage/operations/file.py [hunk 7]
```diff
@@ -407,9 +414,34 @@ def _make_directory_in_files_share(file_service, file_share, directory_path, exi
...
+        return False
+
+
+def file_updates(client, **kwargs):
+    return client.set_http_headers(**kwargs)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/operations/file.py:447]: If no custom logic, just directly use `storage_command`

File: src/azure-cli/azure/cli/command_modules/storage/commands.py [hunk 8]
```diff
@@ -698,9 +685,21 @@ def get_custom_sdk(custom_module, client_factory, resource_type=ResourceType.DAT
...
+        g.storage_command('delete', 'delete_file', transform=create_boolean_result_output_transformer('deleted'),
+                          table_transformer=transform_boolean_for_table)
+        g.storage_command('resize', 'resize_file')
+        g.storage_custom_command('url', 'create_file_url', transform=transform_url_without_encode,
+                                 client_factory=cf_share_client)
+        g.storage_custom_command('generate-sas', 'generate_sas_file', client_factory=cf_share_client)
+        g.storage_command('show', 'get_file_properties', transform=transform_file_show_result,
+                          table_transformer=transform_file_output,
+                          exception_handler=show_exception_handler)
+        g.storage_custom_command('update', 'file_updates', resource_type=ResourceType.DATA_STORAGE_FILESHARE)
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/commands.py:701]:
```suggestion
        g.storage_command('update', 'set_http_headers', resource_type=ResourceType.DATA_STORAGE_FILESHARE)
```

File: src/azure-cli/azure/cli/command_modules/storage/operations/file.py [hunk 9]
```diff
@@ -407,9 +414,34 @@ def _make_directory_in_files_share(file_service, file_share, directory_path, exi
...
+# pylint: disable=redefined-builtin
+def generate_sas_file(cmd, client, directory_name=None, file_name=None, permission=None, expiry=None, start=None,
+                      id=None, ip=None, protocol=None, **kwargs):
+    t_generate_file_sas = get_sdk(cmd.cli_ctx, ResourceType.DATA_STORAGE_FILESHARE,
+                                  '_shared_access_signature#generate_file_sas')
+    file_path = file_name
+    if directory_name:
+        file_path = directory_name + '/' + file_path
+    file_path = file_path.split('/')
+    return t_generate_file_sas(account_name=client.account_name, share_name=client.share_name, file_path=file_path,
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/storage/operations/file.py:433]: Do we need `quote()` before return?

File: src/azure-cli/azure/cli/command_modules/storage/operations/file.py (hunk 7 above)
Comment (calvinhzy) [reply src/azure-cli/azure/cli/command_modules/storage/operations/file.py:447]: This is to fix the default parameter not allowed for update commands lint rule.

File: src/azure-cli/azure/cli/command_modules/storage/operations/file.py (hunk 9 above)
Comment (calvinhzy) [reply src/azure-cli/azure/cli/command_modules/storage/operations/file.py:433]: It is quoted by default.

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**